src/gap_filler.py

```diff
@@ -187,14 +215,75 @@ def function_word_filler(self, txt, removal_proportion=15):
     
         print('\nAnswer key:\n')
    
-        for x in list_of_words:
-            print(f'({answer_key_counter})', x, '\n', )
+        for word in list_of_words:
+            print(f'({answer_key_counter})', word, '\n', )
             answer_key_counter +=1
+
             
             
-            
-            
-            
+    def skim_reader(self, txt):
+
+        doc = self.nlp(txt)
+
+        text_with_gaps = []
+
+        for token in doc:
+            if token.is_stop == False:
+                text_with_gaps.append(token.text)
+            else:
+                text_with_gaps.append("_" * len(token.text))
+
+        print("\n--------> SKIM READER <--------")
+
+        print("\nPractise your skim reading. Read the below text - some words have been removed! What is the general topic of the article?\n")
+
+        print(" ".join(text_with_gaps))
+
+        print("\n")
+
+
+
+    def lemmatizer(self, txt):
```

Comment:
Again looks good but maybe we could do this as a proportion of the sentence.

Also in the future when I get the PDF thing working we might want to make the lemmatised words bold or italic or something. I don't know how to do that yet so lets not worry about it now but in code we usually use 'todo' for these kind of reminders. e.g.:

`#todo make this tand out in pdfs`

Reply:
This reply is in response to all your comments above which refer to removal proportions, etc. Yes, I see what you mean about the possibility for every random number between 0 and target_words potentially being greater than the chosen proportion of target words. Hmm. I'm thinking of somehow adding the target words to a list, and then removing/replacing x number of these from the list. Or something like that. Basically, I guess we just need to move away from random number generators and move towards ensuring a more exact proportion of the target words are always removed. (I'm sure the math.ceil function might still come in handy!) I'll think about it a bit more, but any ideas would be appreciated!

And yeah, I agree, if we could whack all this in its own function it would make it a lot easier if we have future amendments, instead of copy/pasting the code for each of our functions in the GapFinder class. I would add it to the skim_reader and lemmatizer functions too.